### PR TITLE
AST-2857 -  scroll id issue

### PR DESCRIPTION
**Pull request by nathanrodrigues2111**
### Description
jira : https://brainstormforce.atlassian.net/browse/AST-2857

### Screenshots
- 

### Types of changes
Bug fix (non-breaking change which fixes an issue) 

### How has this been tested?
- Checked in frontend with normal link with # and with only hash 

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: assets/js/unminified/frontend.js [hunk 1]
```diff
@@ -1029,8 +1029,13 @@ astScrollToTopHandler = function ( masthead, astScrollTop ) {
...
 		}
 
+		function isUrl(s) {
+			const regexp = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/
+			return regexp.test(s);
+		 }
+
 		function scrollToIDHandler(e) {
-			e.preventDefault();
+			! isUrl ?  e.preventDefault() : '';
```
Comment (imnavanath) [inline assets/js/unminified/frontend.js:1038]: Lets avoid next code execution as well. exectute only `if {}` matches.  wrap the function in if

Comment (imnavanath) [reply assets/js/unminified/frontend.js:1038]: Also add a changelog for this.

Comment (nathanrodrigues2111) [reply assets/js/unminified/frontend.js]: Changelogs added. 

Changing the if condition did not make any change when minifying  @imnavanath

**Review by er-roshan — APPROVED** [review]

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+v4.0.1
+- Fix : Scroll to id issue with anchor links.
```
Comment (imnavanath) [inline changelog.txt:2]:
```suggestion
- Fix : Scroll to Id - Navigation hash link which redirect to other page's section is not clickable.
```

**Review by imnavanath — APPROVED** [review]

**Pull request merged**